Compression.md

```diff
@@ -0,0 +1,145 @@
+
+The specification also discusses the DEFLATE algorithm. We adopt the [zlib](https://www.zlib.net) for doing raw compression and decompression. A pair of a compressor and a decompressor must be set up at both the ends of the connection. The server's decompressor decompresses messages compressed by the client's compressor and vice versa.
+
+### 3. An implementation of permessage-deflate based on swift-nio
+The [swift-nio](https://github.com/apple/swift-nio) framework provides an API which enables HTTP/WebSocket server implementations view the processing of data, read from or written to sockets, as a sequence of transformations that happen through a pipeline of handlers. An active connection is represented by a `Channel`. Data which is read from, or written to, a channel moves through a `ChannelPipeline` of inbound and outbound `ChannelHandlers`. An `EventLoop` is associated with every `Channel`. An `EventLoop`s is a thread-safe abstraction of a thread and provides features for asynchronous code execution using `EventLoopFutures` and `EventLoopPromises`.
+
+In [Kitura-NIO](https://github.com/IBM-Swift/Kitura-NIO), we start the HTTP server with the pipeline configured by swift-nio, adding Kitura-NIO's `HTTPRequestHandler` at the end. A view of the inbound and outbound pipelines (with some handlers omitted for simplicity) is this:
+
+ - Inbound channel handler pipeline: 
+
+    *(Operating System)*   
+         \|      
+    `NIOSSLServerHandler`    
+         \|  
+    `HTTPDecoder`     
+         \|  
+    `HTTPRequestHandler`    
+         \|   
+    *(Kitura/WebSocket app)*  
+    
+ - Outbound channel handler pipeline: 
+
+    *(Kitura/WebSocket app)*  
+    \|  
+    `HTTPEncoder`  
+    \|  
+    `NIOSSLServerHandler`  
+    \|  
+    *(Operating System)*   
+
+[HTTPDecoder](https://apple.github.io/swift-nio/docs/current/NIOHTTP1/Classes/HTTPDecoder.html) and [HTTPResponseEncoder](https://apple.github.io/swift-nio/docs/current/NIOHTTP1/Classes/HTTPResponseEncoder.html) convert bytes to HTTP requests, and responses to bytes, respectively. [NIOSSLServerHandler](https://apple.github.io/swift-nio-ssl/docs/current/NIOSSL/Classes/NIOSSLServerHandler.html) is a duplex handler (both inbound and outbound) used to decrypt and encrypt data on a secure connection. The [HTTPRequestHandler](https://github.com/IBM-Swift/Kitura-NIO/blob/master/Sources/KituraNet/HTTP/HTTPRequestHandler.swift) is used to invoke Kitura's router.
```

Comment:
Why is the HTTPRequestHandler link the only one that doesn't link to API doc?

Reply:
The `HTTPRequestHandler` is not a part of the API. It's an internal class.